File: CppChatClient/message/PTextMessageContent.cpp
//
//  PTextMessageContent.cpp
//  ChatClient
//
//  Created by Hao Jia on 2020/1/31.
//  Copyright © 2020 WF Chat. All rights reserved.
//

#include "PTextMessageContent.h"

namespace WFCLib {

PTextMessageContentPrototype PTextMessageContent::sPrototype;

MessagePayload PTextMessageContent::encode() const {
    MessagePayload payload = MessageContent::encode();
    payload.searchableContent = content;
    return payload;
}


void PTextMessageContent::decode(const MessagePayload & payload){
    MessageContent::decode(payload);
    content = payload.searchableContent;
}

MessageContent* PTextMessageContent::clone() const {
    PTextMessageContent *p = new PTextMessageContent();
    p->extra = extra;
    p->content = content;
    return p;
}

const MessageContentPrototype* PTextMessageContent::getPrototype() const {
    return &sPrototype;
}

std::string PTextMessageContent::digest() const {
    return content;
}
};

File: CsChatClient/Messages/PTextMessageContent.cs
namespace CsChatClient.Messages
{
    [ContentAttribute(MessageContentType.MESSAGE_CONTENT_TYPE_P_TEXT, MessageContentPersistFlag.PersistFlag_PERSIST_AND_COUNT)]
    public class PTextMessageContent : MessageContent
    {
        public string Content { get; set; }

        public override void Decode(MessagePayload payload)
        {
            Content = payload.Content;
        }

        public override string Digest(MessageEx message)
        {
            return Content;
        }

        public override MessagePayload Encode()
        {
            MessagePayload payload = new MessagePayload();
            payload.Content = Content;
            return payload;
        }
    }
}
